lldb/source/Plugins/SymbolLocator/CMakeLists.txt

```diff
@@ -0,0 +1,4 @@
+add_subdirectory(Default)
+if (CMAKE_SYSTEM_NAME MATCHES "Darwin")
+  add_subdirectory(DebugSymbols)
```

Comment:
I'm switching my previous PR to be a "Debuginfod" SymbolLocator plugin, and I think this plugin name is far too general. @clayborg tells me that  "DebugSymbols.framework" is a thing on MacOS, but maybe rename this "DebugSymbolsApple" or "MacOSDebugSymbols" or pretty much anything that would indicate to the random developer that this stuff is for (mac|i(Pad)|tv)OS/Apple/Darwin/Sherlock/dSYM symbol lookup.